File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;

    public int solution(int n, int[][] computers) {
        visited = new boolean[n];
        int answer = 0;

        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                answer++;
                bfs(computers, i);
            }
        }
        
        return answer;
    }
    
    public static void bfs(int[][] computers, int vertex) {
        Queue<Integer> q = new LinkedList<>();
        q.add(vertex);

        while (!q.isEmpty()) {
            Integer now = q.poll();
            visited[now] = true;

            for (int i = 0; i < computers[now].length; i++) {
                if (computers[now][i] == 1 && !visited[i] && !q.contains(i)) {
                    q.add(i);
                }
            }
        }
    }

}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque

def bfs(start, computers, visited):
    answer = 0
    queue = deque([start])
    while queue:
        now = queue.popleft()
        if now not in visited:
            answer += 1
            visited.append(now)
        for i in range(len(computers)):
            if i != now and computers[now][i] == 1 and i not in visited:
                visited.append(i)
                queue.append(i)
    return answer

def solution(n, computers):
    answer = 0
    visited = []
    for i in range(n):
        answer += bfs(i, computers, visited)
    return answer
